### ISSUE TITLE
regression: new windows runners produce malfunctioning binaries

### ISSUE THREAD

**Issue by Green-Sky** [actions/runner-images#9038]
### Description

Starting 2 days ago, new generated binaries, even of older previously working commit, dont seems to be able to properly establish connections to the tox network. I am still digging into the code base to find out more about where the issue really is, but since the only change that seems to make a difference is the fact that the "binary is new", and I need to run this through the ci, this is somewhat of a chore.

looking at the runner update notes the only thing that might cause issues i the vcpkg update: https://github.com/microsoft/vcpkg/compare/08c4e7104...000d1bda1
but I cant fine anything in there that might cause issues. (pkg-config is not used on windows)

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [X] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

Image: windows-2022
Version: 20231211.1.0
AND
Image: windows-2019
Version: 20231211.1.0

### Is it regression?
Yes, last successful run:
Image: windows-2022 Version: 20231205.1.0 https://github.com/Green-Sky/tomato/actions/runs/7207181708/job/19633517719

### Expected behavior

the produced binary should start and connect to the tox network.

### Actual behavior

The produced binary renders normally, and constantly produces network traffic, but cant connect to the network.

### Repro steps

(see expected vs actual)

**Comment by shamil-mubarakshin** [comment]
Hello @Green-Sky.
Saw latest runs are green. Is issue still persists and could you provide link to the failing run?

**Comment by Green-Sky** [comment]
Sorry for the lack of updates. I spend hours trying to pin this down. I am no insane.
I am still unsure what is the actual cause, or why the runner update causes this.

First I did printf debugging with the ci (~5min turnaround time). I figured out that executing the binaries on my system using wine exhibits the same issues.
I figured out that the timer in toxcore was spewing out random numbers, instead of timestamps.
Keep in mind that this is specific to the runner version, and the same commits with previous runner versions produce correctly behaving binaries, so it is not specific to the environment it is executed it.

I then tried to add a test case to produce the bug in a somewhat isolated code environment, but still inside the same repository and discovered that THE MERE PRESENCE of the "new" source file, that is not compiled, causes the binary to be correct again. [diff with messy commit, only look at the actual file changes](https://github.com/Green-Sky/tomato/compare/4dca80c...4cf5dad).
I also discovered that compiling the timer as c++ instead of c fixes the binaries too. (mono_timer.c -> mono_timer.cpp + cmake change)

It is now the 3rd day and over 30 ci artifacts since I discovered this and I am turning insane.

(update:) While writing this update, [I realized that an older commit that was rerun on the new runner images works 7495a50 ](https://github.com/Green-Sky/tomato/actions/runs/7240834154), but [the next commit following does not 6b5f40b
](https://github.com/Green-Sky/tomato/actions/runs/7233525179)

Here is a rundown on what changes "fixed" the regression:
- changing mono_time.c to be compiled as c++
- adding a new source file that is not even compiled

changes that did not fix the regression:
- using old commits (in almost all cases, exception seen above)
- using new commits (updating the toxcore subtree, which mono_time.c is part of)
- changing related toxcore files (added print statements)
- changing the runner from windows-2022 to windows-2019


I am sorry for opening this issue here, feel free to close it, if you dont see the relevance to runner-images :).
It has been a while since I have felt this lost in the dark... Time to move do binary diffing.
